lib/libc/string/memchr.c

```diff
@@ -33,14 +33,24 @@ __FBSDID("$FreeBSD$");
 #define ONES ((size_t)-1 / UCHAR_MAX)
 #define HIGHS (ONES * (UCHAR_MAX / 2 + 1))
 #define HASZERO(x) (((x)-ONES) & ~(x)&HIGHS)
+#define MIN(a, b) ((a) < (b) ? (a) : (b))
 
 void *
-memchr(const void *src, int c, size_t n)
+memchr(const void *src, int c, size_t len)
```

Comment:
C11 states the prototype `void *memchr(const void *s, int c, size_t n);` so having `n` mean something else could be confusing. I think I'd prefer the word loop to use a different variable name unless you have a compelling reason why you chose this way? Also `len` isn't great as a name given capabilities have a length...

Reply:
I chose this it minimize diffs to upstream since `n` is used on many lines.  I certainly couple pick something else.

Reply:
I've got a rework that avoids all these issues, will push once it's tested.